spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -210,33 +253,47 @@ In plain English, between chains `A` and `B`:
   an acknowledgement of failure is preferable to aborting the transaction since it more easily enables the sending chain
   to take appropriate action based on the nature of the failure.
 
+Note: `constructOnChainDenom` is a helper function that will construct the local on-chain denomination for the bridged token. It **must** encode the trace and base denomination to ensure that tokens coming over different paths are not treated as fungible. The original trace and denomination must be retrievable by the state machine so that they can be passed in their original forms when constructing a new IBC path for the bridged token. The ibc-go implementation handles this by creating a local denomination: `hash(trace+base_denom)`.
+
```

Comment:
Maybe we could also add a notes for functions `getSentMetadataFlag` and `getMetadata`.

Reply:
The explanation of these functions is at line 490 under the populating token metadata section.